prov/sockets/src/sock_progress.c

```diff
@@ -2453,6 +2456,65 @@ static void sock_pe_poll(struct sock_pe *pe)
 	}
 }
 
+#ifndef __APPLE__
+static void sock_thread_set_affinity(char *s)
+{
+	char *saveptra, *saveptrb, *saveptrc;
+	char *a, *b, *c;
+	int j, first, last,stride;
+	cpu_set_t mycpuset;
+	pthread_t mythread;
+
+	mythread = pthread_self();
+	CPU_ZERO(&mycpuset);
+
+	a = strtok_r(s,",", &saveptra);
+	while (a)
+	{
+		first = last = -1;
+		stride = 1;
+		b = strtok_r(a,"-",&saveptrb);
+		first = atoi(b);
+		/* Check for range delimiter */
+		if ((b = strtok_r(NULL,"-",&saveptrb)))
+		{
+			c = strtok_r(b,":",&saveptrc);
+			last = atoi(c);
+			/* Check for stride */
+			if ((c = strtok_r(NULL,":",&saveptrc))) {
+				stride = atoi(c);
+			}
+		}
+		
+		if (last == -1)
+			last = first;
+		
+		for (j = first; j<= last; j+=stride)
+			CPU_SET(j, &mycpuset);
+		a =  strtok_r(NULL,",", &saveptra);
+	}
+	
+	j = pthread_setaffinity_np(mythread, sizeof(cpu_set_t), &mycpuset);
+	if (j != 0) 
+		SOCK_LOG_ERROR("pthread_setaffinity_np failed\n");
+}
+#endif
+
+static void sock_pe_set_affinity (void)
+{
+	char *s;
+	fi_var_register(&sock_prov, "pe_affinity",
+			"Which logical core(s) to be used for binding the progress thread. This option is currently not supported on OS X. Usage: virtual_processor_id_start[-virtual_processor_id_end[:stride]][,<next_rule>]");
```

Comment:
Maybe:

"If specified, bind the progress thread to the indicatedrange(s) of Linux virtual processor ID(s). This option is currently not supported on OS X. Usage: id_start[-id_end[:stride]][,<next_range>]"